utils/errors.go

```diff
@@ -0,0 +1,15 @@
+package utils
+
+import "fmt"
+
+func MergeErrorsOrNil(errors []error, description string) error {
```

Comment:
Why not create custom type errors that will implement error interface and do all that logic there.